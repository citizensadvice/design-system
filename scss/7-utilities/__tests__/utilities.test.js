/* eslint-env jest */
const glob = require('glob');
const sass = require('sass');

test.each(glob.sync('scss/7-utilities/_*.scss'))(
  '%p can be imported standalone',
  (file) => {
    expect(() =>
      sass.renderSync({
        includePaths: ['scss/'],
        file,
      }),
    ).not.toThrow();
  },
);

test('visibility styles match snapshot', () => {
  const output = sass.renderSync({
    file: 'scss/7-utilities/_visibility.scss',
  });
  expect(output.css.toString()).toMatchSnapshot();
});
