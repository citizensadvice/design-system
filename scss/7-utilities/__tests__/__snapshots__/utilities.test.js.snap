// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`visibility styles match snapshot 1`] = `
".cads-hide {
  display: none !important;
}

.cads-invisible {
  visibility: hidden !important;
}

.cads-visually-hidden,
.cads-sr-only {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.cads-sr-only-focusable:active, .cads-sr-only-focusable:focus {
  position: static;
  width: auto;
  height: auto;
  overflow: visible;
  clip: auto;
  white-space: normal;
}

.cads-show-md-up {
  display: none !important;
}
@media (min-width: 768px) {
  .cads-show-md-up {
    display: inherit !important;
  }
}

.cads-show-md-down {
  display: inherit !important;
}
@media (min-width: 768px) {
  .cads-show-md-down {
    display: none !important;
  }
}

.cads-show-md-only {
  display: none !important;
}
@media (min-width: 768px) and (max-width: 1023.98px) {
  .cads-show-md-only {
    display: inherit !important;
  }
}

@media (min-width: 768px) {
  .cads-hide-md-up {
    display: none !important;
  }
}

.cads-hide-md-down {
  display: none !important;
}
@media (min-width: 768px) {
  .cads-hide-md-down {
    display: inherit !important;
  }
}

@media (min-width: 768px) and (max-width: 1023.98px) {
  .cads-hide-md-only {
    display: none !important;
  }
}

.cads-show-lg-up {
  display: none !important;
}
@media (min-width: 1024px) {
  .cads-show-lg-up {
    display: inherit !important;
  }
}

.cads-show-lg-down {
  display: inherit !important;
}
@media (min-width: 1024px) {
  .cads-show-lg-down {
    display: none !important;
  }
}

.cads-show-lg-only {
  display: none !important;
}
@media (min-width: 1024px) {
  .cads-show-lg-only {
    display: inherit !important;
  }
}

@media (min-width: 1024px) {
  .cads-hide-lg-up {
    display: none !important;
  }
}

.cads-hide-lg-down {
  display: none !important;
}
@media (min-width: 1024px) {
  .cads-hide-lg-down {
    display: inherit !important;
  }
}

@media (min-width: 1024px) {
  .cads-hide-lg-only {
    display: none !important;
  }
}"
`;
